src/formats/PDB.cpp

```diff
@@ -639,6 +639,102 @@ static std::string to_pdb_index(uint64_t i) {
     }
 }
 
+struct ResidueInformation {
+    std::string atom_hetatm{ "HETATM" };
+    std::string resname { "XXX" };
+    std::string resid { "  -1" };
+    std::string chainid { "X" };
+    std::string inscode { " " };
+    std::string comp_type { "" };
+};
+
+static ResidueInformation get_residue_strings(optional<const Residue&> residue_opt, uint64_t& max_resid) {
+
+    ResidueInformation res_info;
+
+    if (!residue_opt) {
+        auto value = max_resid++;
+        if (value < 9999) {
+            res_info.resid = to_pdb_index(value);
+        }
+        else {
+            res_info.resid = "  -1";
```

Comment:
Should we emit a warning? 

Also, since we are touching this code, should we use some kind of hexadecimal numbering scheme if we have more than 9999 residues? I know that's what VMD is doing for such "non standard" files: https://www.ks.uiuc.edu/Research/vmd/mailing_list/vmd-l/7384.html

Also, do we want to use `-1` or another special value, such as `*****`/`0`/`    `?

Reply:
the hexadecimal thing have a standard: http://cci.lbl.gov/hybrid_36/

Reply:
I think this is a great idea! They provide sample code as well and it appears PHENIX (a popular program for manipulating PDBs) supports this standard. I'll check to see if PyMOL supports the standard, but it looks unlikely (maybe I'll add it there as well).

> Also, do we want to use -1 or another special value, such as `*****`/`0`/` `?

I'm not sure. I'll see what other software supports. My biggest worry is that non-integer characters will cause problems with other software.